src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,184 @@
+        preparedStatement.setLong(2, roleId);
+        return preparedStatement.executeUpdate();
+    }
+
+    @Override
+    public Optional<User> get(Long id) {
+        String query = "SELECT *, group_concat(roles.role_name SEPARATOR ',') AS role_titles"
+                        + " FROM users"
+                        + " JOIN users_roles ON users.user_id = users_roles.user_id"
+                        + " JOIN roles ON users_roles.role_id = roles.role_id"
+                        + " where users.user_id = ?";
+
+        try (Connection connection = DbConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            ResultSet sqlResult = preparedStatement.executeQuery();
+            if (sqlResult.next()) {
+                User user = parseRow(sqlResult);
+                return Optional.of(user);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get user by id = " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    private User parseRow(ResultSet sqlResult) throws SQLException {
+        Long id = sqlResult.getLong("user_id");
+        String name = sqlResult.getString("user_name");
+        String login = sqlResult.getString("login");
+        String password = sqlResult.getString("password");
+        String[] rolesStringArr = sqlResult.getString("role_titles").split(",");
+        Set<Role> roles = new HashSet();
+        for (String str : rolesStringArr) {
+            roles.add(Role.of(str));
+        }
+        return new User(id, name, login, password, roles);
+    }
+
+    @Override
+    public User update(User user) {
+        String query = "UPDATE users"
+                + " SET user_name = ?, password = ?, login = ? "
+                + " WHERE user_id = ?";
+
+        try (Connection connection = DbConnectionUtil.getConnection()) {
+            PreparedStatement prepareStatement = connection.prepareStatement(query);
+            prepareStatement.setString(1, user.getName());
+            prepareStatement.setString(2, user.getPassword());
+            prepareStatement.setString(3, user.getLogin());
+            prepareStatement.setLong(4, user.getId());
+            prepareStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("User with id " + user.getId()
+                    + " has not been updated", e);
+        }
+        return user;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE users"
+                + " SET deleted = true "
+                + " WHERE user_id = ?";
+
+        try (Connection connection = DbConnectionUtil.getConnection()) {
+            PreparedStatement prepareStatement = connection.prepareStatement(query);
+            prepareStatement.setLong(1, id);
+            prepareStatement.executeUpdate();
```

Comment:
you can compare the result of ... .executeUpdate with 0 , i.e. set 
int smth = 0, 
smth = prepareStatement.executeUpdate;
return smth > 0;